Four playback issues mainly from discord rpc

**Describe the bug**
Stacktrace from the hell machine named chatgpt

Discord RPC crash with current pypresence
Error: TypeError: Presence.update() got an unexpected keyword argument 'name'
Where: high_tide/lib/discord_rpc.py in set_activity()
When: on play, on track start, and on seek
Impact: uncaught exception in playback paths; can break normal playback flow.

Playback UI timer/slider breaks after Discord RPC exception
Symptom: progress slider shows 0:00 / 0:00 while MPRIS position still updates correctly.
Cause chain: exception thrown before/inside playback update path, so UI slider/timer updates do not run reliably.
Impact: UI desync (audio may continue, UI progress does not).

Next track may not start when stream fetch fails
Error: 500 Server Error from TIDAL endpoint .../playbackinfopostpaywall
Where: track.get_stream() in player_object.py
Symptom: current/next track fails to load and playback can stall instead of advancing.
Impact: auto-advance can stop when one track fetch fails.

Auto-next regression visible after seek-near-end + failures
Log clue: Ignoring about to finish event
Observed behavior: after song end, next song sometimes does not start.
Likely trigger: combination of near-end seek state + stream/RPC errors.



**Expected behavior**

For the playback to not stall

**System:**
 - OS: EndeavourOS Arch-based
 - Package: Flatpak
 - Version 1.2.0

**Additional context**

Logs

```
poco@archlinux ~> flatpak run io.github.nokse22.high-tide
2026-03-09 14:13:49,836 [INFO] high_tide.lib.player_object: GStreamer version: GStreamer 1.26.10
2026-03-09 14:13:49,844 [INFO] high_tide.window: Proxy enabled for network requests
2026-03-09 14:13:49,847 [INFO] high_tide.lib.secret_storage: initializing secret store
2026-03-09 14:13:52,435 [INFO] high_tide.lib.utils: Favorite Artists: 0
2026-03-09 14:13:52,435 [INFO] high_tide.lib.utils: Favorite Tracks: 0
2026-03-09 14:13:52,435 [INFO] high_tide.lib.utils: Favorite Albums: 0
2026-03-09 14:13:52,435 [INFO] high_tide.lib.utils: Favorite Playlists: 1
2026-03-09 14:13:52,435 [INFO] high_tide.lib.utils: Favorite Mixes: 0
2026-03-09 14:13:52,435 [INFO] high_tide.lib.utils: Playlist and Favorite Playlists: 1
2026-03-09 14:13:52,435 [INFO] high_tide.lib.utils: User Playlists: 1
2026-03-09 14:13:52,435 [INFO] high_tide.window: logged in
2026-03-09 14:13:52,439 [INFO] high_tide.window: Last playing: af450503-7072-466d-bf2a-0ad8a0b4c027 of type playlist index: 326
2026-03-09 14:13:53,898 [INFO] high_tide.lib.player_object: file:///home/poco/.var/app/io.github.nokse22.high-tide/cache/manifest.mpd
2026-03-09 14:13:53,899 [INFO] high_tide.window: song changed
2026-03-09 14:13:55,074 [INFO] high_tide.lib.discord_rpc: Connected to discord IPC
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 596, in on_play_button_clicked
    self.player_object.play_pause()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 441, in play_pause
    self.play()
    ~~~~~~~~~^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 421, in play
    discord_rpc.set_activity(
    ~~~~~~~~~~~~~~~~~~~~~~~~^
        self.playing_track, self.query_position() / 1_000_000_000
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 321, in _on_track_start
    discord_rpc.set_activity(self.playing_track, 0)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:13:56,857 [INFO] tidalapi.request: Request resulted in exception 500 Server Error: Internal Server Error for url: https://api.tidal.com/v1/tracks/1269502/playbackinfopostpaywall?sessionId=40996a36-9091-44e1-919b-608ac0fd0330&countryCode=BR&limit=1000&playbackmode=STREAM&audioquality=HI_RES_LOSSLESS&assetpresentation=FULL
2026-03-09 14:13:56,857 [ERROR] high_tide.lib.player_object: Error getting track URL
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 464, in _play_track_thread
    self.stream = track.get_stream()
                  ~~~~~~~~~~~~~~~~^^
  File "/app/lib/python3.13/site-packages/tidalapi/media.py", line 449, in get_stream
    request = self.requests.request(
        "GET", "tracks/%s/playbackinfopostpaywall" % self.id, params
    )
  File "/app/lib/python3.13/site-packages/tidalapi/request.py", line 153, in request
    request.raise_for_status()
    ~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/app/lib/python3.13/site-packages/requests/models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://api.tidal.com/v1/tracks/1269502/playbackinfopostpaywall?sessionId=40996a36-9091-44e1-919b-608ac0fd0330&countryCode=BR&limit=1000&playbackmode=STREAM&audioquality=HI_RES_LOSSLESS&assetpresentation=FULL
2026-03-09 14:13:59,808 [INFO] tidalapi.request: Request resulted in exception 500 Server Error: Internal Server Error for url: https://api.tidal.com/v1/tracks/600179/playbackinfopostpaywall?sessionId=40996a36-9091-44e1-919b-608ac0fd0330&countryCode=BR&limit=1000&playbackmode=STREAM&audioquality=HI_RES_LOSSLESS&assetpresentation=FULL
2026-03-09 14:13:59,808 [ERROR] high_tide.lib.player_object: Error getting track URL
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 464, in _play_track_thread
    self.stream = track.get_stream()
                  ~~~~~~~~~~~~~~~~^^
  File "/app/lib/python3.13/site-packages/tidalapi/media.py", line 449, in get_stream
    request = self.requests.request(
        "GET", "tracks/%s/playbackinfopostpaywall" % self.id, params
    )
  File "/app/lib/python3.13/site-packages/tidalapi/request.py", line 153, in request
    request.raise_for_status()
    ~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/app/lib/python3.13/site-packages/requests/models.py", line 1026, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://api.tidal.com/v1/tracks/600179/playbackinfopostpaywall?sessionId=40996a36-9091-44e1-919b-608ac0fd0330&countryCode=BR&limit=1000&playbackmode=STREAM&audioquality=HI_RES_LOSSLESS&assetpresentation=FULL
2026-03-09 14:14:04,881 [INFO] high_tide.window: seeking: 0.9535398230088495
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:04,910 [INFO] high_tide.window: seeking: 0.9601769911504425
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:04,935 [INFO] high_tide.window: seeking: 0.9778761061946902
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:04,960 [INFO] high_tide.window: seeking: 0.9867256637168141
2026-03-09 14:14:04,973 [INFO] high_tide.window: seeking: 0.013274336283185861
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:04,986 [INFO] high_tide.window: seeking: 0.013274336283185861
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:05,001 [INFO] high_tide.window: seeking: 0.013274336283185861
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:05,027 [INFO] high_tide.window: seeking: 0.013274336283185861
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:05,040 [INFO] high_tide.window: seeking: 0.013274336283185861
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:05,057 [INFO] high_tide.window: seeking: 0.013274336283185861
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:05,120 [INFO] high_tide.window: seeking: 0.013274336283185861
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:05,162 [INFO] high_tide.lib.player_object: https://lgf.audio.tidal.com/mediatracks/CAEaKwgDEidkYjBiMDQyMTNlOTZhOTZkOWVlYWM5YmY0YWI0NzU2N182MS5tcDQ/0.flac?token=1773058161~NjlmZDY5NzAxZDczMGZkZWYzOGUwYjllMjk4MjMyN2Q2NWQ5NjRmMw==
2026-03-09 14:14:05,163 [INFO] high_tide.window: song changed
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 548, in _play_track_url
    self.play()
    ~~~~~~~~~^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 421, in play
    discord_rpc.set_activity(
    ~~~~~~~~~~~~~~~~~~~~~~~~^
        self.playing_track, self.query_position() / 1_000_000_000
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 321, in _on_track_start
    discord_rpc.set_activity(self.playing_track, 0)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:07,869 [INFO] high_tide.window: seeking: 0.013274336283185861
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:07,917 [INFO] high_tide.window: seeking: 0.09734513274336276
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:07,961 [INFO] high_tide.window: seeking: 0.09734513274336276
Traceback (most recent call last):
  File "/app/share/high-tide/high_tide/window.py", line 670, in on_slider_seek
    self.player_object.seek(seek_fraction)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/player_object.py", line 742, in seek
    discord_rpc.set_activity(self.playing_track, position // 1_000_000_000)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/high-tide/high_tide/lib/discord_rpc.py", line 123, in set_activity
    rpc.update(
    ~~~~~~~~~~^
        activity_type=ActivityType.LISTENING,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        end=int(time.time() - offset_ms + track.duration),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
TypeError: Presence.update() got an unexpected keyword argument 'name'
2026-03-09 14:14:36,796 [INFO] high_tide.lib.player_object: Ignoring about to finish event
^C⏎                                                                                                                                                                                                                                         poco@archlinux ~ [SIGINT]> 

```

--- a/src/lib/discord_rpc.py
+++ b/src/lib/discord_rpc.py
@@ -100,10 +100,9 @@ def set_activity(track: Track | None = None, offset_ms: int = 0) -> None:
         try:
             if state != State.DISCONNECTED:
                 rpc.clear()
-                rpc.close()
         except Exception:
             pass
-        state = State.DISCONNECTED
+        state = State.IDLE
         return
 
     if state == State.DISCONNECTED:
@@ -123,12 +122,19 @@ def set_activity(track: Track | None = None, offset_ms: int = 0) -> None:
             rpc.update(
                 activity_type=ActivityType.LISTENING,
                 details=track.name,
-                name=", ".join(artists) if artists else "Unknown Artist",
-                state=", ".join(artists) if artists else "Unknown Artist",
-                large_image=track.album.image() if track.album else "hightide_x1024",
-                small_text="High Tide" if track.album else None,
+                state=", ".join(artists)
+                if artists
+                else "Unknown Artist",
+                large_image=track.album.image()
+                if track.album
+                else "hightide_x1024",
+                small_text="High Tide"
+                if track.album
+                else None,
                 start=int(time.time() - offset_ms),
-                end=int(time.time() - offset_ms + track.duration),
+                end=int(
+                    time.time() - offset_ms + track.duration
+                ),
             )
             state = State.PLAYING
     except pypresence.exceptions.PipeClosed:

--- a/src/lib/player_object.py
+++ b/src/lib/player_object.py
@@ -750,7 +750,8 @@ class PlayerObject(GObject.GObject):
         self.discord_rpc_enabled = enabled
         if enabled and self.playing:
             discord_rpc.set_activity(
-                self.playing_track, int(self.query_duration()) // 1_000_000_000
+                self.playing_track,
+                self.query_position() / 1_000_000_000,
             )
         elif enabled:
             discord_rpc.set_activity()

